src/test/metrics_api_test.cc

```diff
@@ -232,6 +233,32 @@ MetricAPIHelper(TRITONSERVER_Metric* metric, TRITONSERVER_MetricKind kind)
   TRITONSERVER_ErrorDelete(err);
 }
 
+void
+HistogramAPIHelper(TRITONSERVER_Metric* metric)
+{
+  // Observe
+  std::vector<double> data{0.05, 1.5, 6.0};
+  std::vector<std::uint64_t> cumulative_counts = {1, 1, 2, 2, 3, 3};
+  double sum = 0.0;
+  for (auto datum : data) {
+    FAIL_TEST_IF_ERR(
+        TRITONSERVER_MetricObserve(metric, datum), "observe metric value");
+    sum += datum;
+  }
+
+  // Collect
+  prometheus::ClientMetric value;
```

Comment:
Leaking implementation..

Reply:
Suggestions?

Reply:
Remove the collect function and check the metrics by parsing from `TRITONSERVER_MetricsFormatted`

Reply:
Updated